### PR TITLE
[shelly] Fix UNI support up to three DS18B20 sensors

### PR DESCRIPTION
**Pull request by bofakin**
# Description

For the Shelly UNI a small fix needs to be applied to the JSON that is received form the device. When two or more temperature sensors are attached, the array is not sorted and the sensor "1" is first, causes the string replacement to fail.

With this patch all tree possible variants are checked, so that the JSON fix is applied for all possible sortings.

Note: The Shelly UNI supports only up to three DS18B20.

# Testing

Here is the build artifact:
https://openhab.jfrog.io/artifactory/libs-pullrequest-local/org/openhab/addons/bundles/org.openhab.binding.shelly/4.1.0-SNAPSHOT/org.openhab.binding.shelly-4.1.0-SNAPSHOT.jar

Installed and tested on my local system - Shelly gets `ONLINE` and all three DS18B20 are detected and working.

Note: This is my first pull request, I hope I'll got everything correct.

### CODE REVIEW

**Review by lsiepel — APPROVED** [review]
LGTM

**Comment by lolodomo** [comment]
@markus7017: could you please review ?

File: bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api/ShellyDeviceProfile.java [hunk 1]
```diff
@@ -126,7 +126,10 @@ public ShellyDeviceProfile initialize(String thingType, String jsonIn) throws Sh
         initFromThingType(thingType);
 
         String json = jsonIn;
-        if (json.contains("\"ext_temperature\":{\"0\":[{")) {
+        // It is not guaranteed, that the array entries are in order. Check all
+        // possible variants. See openhab#15514.
+        if (json.contains("\"ext_temperature\":{\"0\":[{") || json.contains("\"ext_temperature\":{\"1\":[{")
+                || json.contains("\"ext_temperature\":{\"2\":[{")) {
```
Comment (lolodomo) [inline bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api/ShellyDeviceProfile.java:132]: Wouldn't this be sufficient ?
```
if (json.contains("\"ext_temperature\":{\"")) {
```

Comment (bofakin) [reply bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api/ShellyDeviceProfile.java:132]: I can't tell.

I assume there was a reason, the JSON string is checked including the array index. So I just added the two other possible cases, but keeping as close to the original condition as possible.

Comment (lolodomo) [reply bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api/ShellyDeviceProfile.java:132]: Let's wait for @markus7017 feedback

**Comment by lolodomo** [comment]
@markus7017 : are you there ?

**Comment by markus7017** [comment]
LGTM 
thanks @bofakin

**Review by lolodomo — APPROVED** [review]

**Pull request merged**